app/src/main/java/com/kickstarter/viewmodels/ProjectHolderViewModel.java

```diff
@@ -0,0 +1,497 @@
+    private final KSString ksString;
+
+    public ViewModel(final @NonNull Environment environment) {
+      super(environment);
+      this.ksCurrency = environment.ksCurrency();
+      this.ksString = environment.ksString();
+
+      final Observable<Project> project = this.projectAndCountry.map(PairUtils::first);
+      final Observable<ProjectUtils.Metadata> projectMetadata = project.map(ProjectUtils::metadataForProject);
+
+      this.avatarPhotoUrl = project.map(p -> p.creator().avatar().medium());
+      this.backersCountTextViewText = project.map(Project::backersCount).map(NumberUtils::format);
+
+      this.backingViewGroupIsGone = projectMetadata
+        .map(ProjectUtils.Metadata.BACKING::equals)
+        .map(BooleanUtils::negate);
+
+      this.blurbTextViewText = project.map(Project::blurb);
+      this.categoryTextViewText = project.map(Project::category).filter(ObjectUtils::isNotNull).map(Category::name);
+
+      this.commentsCountTextViewText = project
+        .map(Project::commentsCount)
+        .filter(ObjectUtils::isNotNull)
+        .map(NumberUtils::format);
+
+      this.creatorNameTextViewText = project.map(p -> p.creator().name());
+      this.deadlineCountdownTextViewText = project.map(ProjectUtils::deadlineCountdownValue).map(NumberUtils::format);
+
+      this.featuredViewGroupIsGone = projectMetadata
+        .map(ProjectUtils.Metadata.CATEGORY_FEATURED::equals)
+        .map(BooleanUtils::negate);
+
+      this.featuredTextViewRootCategory = this.featuredViewGroupIsGone
+        .filter(BooleanUtils::isFalse)
+        .compose(combineLatestPair(project))
+        .map(bp -> bp.second.category())
+        .filter(ObjectUtils::isNotNull)
+        .map(Category::root)
+        .filter(ObjectUtils::isNotNull)
+        .map(Category::name);
+
+      this.goalStringForTextView = project
+        .map(p -> this.ksCurrency.format(p.goal(), p, false, true, RoundingMode.DOWN));
+
+      this.locationTextViewText = project
+        .map(Project::location)
+        .filter(ObjectUtils::isNotNull)
+        .map(Location::displayableName);
+
+      this.percentageFundedProgress = project.map(Project::percentageFunded).map(ProgressBarUtils::progress);
+
+      this.percentageFundedProgressBarIsGone = project
+        .map(p -> p.isSuccessful() || p.isCanceled() || p.isFailed() || p.isSuspended());
+
+      this.playButtonIsGone = project.map(Project::hasVideo).map(BooleanUtils::negate);
+
+      this.pledgedTextViewText = project
+        .map(p -> this.ksCurrency.format(p.pledged(), p, false, true, RoundingMode.DOWN));
+
+      this.potdViewGroupIsGone = projectMetadata
+        .map(ProjectUtils.Metadata.POTD::equals)
+        .map(BooleanUtils::negate);
+
+      this.projectDisclaimerGoalReachedDateTime = project
+        .filter(Project::isFunded)
+        .map(Project::deadline);
+
+      this.projectDisclaimerGoalNotReachedString = project
+        .filter(p -> p.deadline() != null && p.isLive() && !p.isFunded())
```

Comment:
what's the case where a deadline would be null?

Reply:
since the deadline comes from the `_mini` and not the `_compact`, serializer, it's a nullable field. also for projects that are in draft mode (fwiw i think we exclude projects in draft or "submitted" state from the api, but this also makes `deadline` technically nullable) :/